### PR TITLE
Notes on implementation; scoping compared to draft conformance rules …

### PR DESCRIPTION
**Pull request by pp-mo**
…+ ugrid-checks.

### CODE REVIEW

**Comment by ocefpaf** [comment]
@pp-mo I'd like to deprecate this one in lieu for `ugrid-checks`, but I'd like to see if we can make `ugrid-checks` work as a compliance-checker plugin first. Probably not an easy feat...

**Comment by pp-mo** [comment]
>  @ocefpaf see if we can make ugrid-checks work as a compliance-checker plugin 

Thanks for the plug !

I did look into wrapping ugrid-checks within the ioos framework before, but only as a preliminary investigation.
It was discussed in https://github.com/ioos/ioos-code-sprint/issues/6
My own ideas notes here : https://github.com/pp-mo/ugrid-checks/issues/44
Meanwhile a draft (partial) code interfacing approach is visible [here](https://github.com/pp-mo/ugrid-checks/compare/main...ioos_category_trial).

I would note that, when I discussed with another person (? @MathewBiddle or someone else ?) in context of the existing code, it seemed that the existing implementation was a more structured validation approach than the ugrid-checks code, which is arguably a bit ad-hoc in it's approach (it was written under some time constraint).  I think you can see that in the ioos-interfacing draft code outlined above .  

I think it's also worth saying that, like the UGRID spec itself, the ugrid-checks code has been pretty much static for some years + gets very little attention, or at least little feedback.
Which is not to say that it doesn't provide some value, and is still functional.

So it does need careful consideration whether somewhat ugrid-checks code, and the above interfacing approach is generally good enough to adopt here -- I will appreciate comment on that from others !

**Comment by ocefpaf** [comment]
> I did look into wrapping ugrid-checks within the ioos framework before, but only as a preliminary investigation. It was discussed in [ioos/ioos-code-sprint#6](https://github.com/ioos/ioos-code-sprint/issues/6) My own ideas notes here : [pp-mo/ugrid-checks#44](https://github.com/pp-mo/ugrid-checks/issues/44) Meanwhile a draft (partial) code interfacing approach is visible [here](https://github.com/pp-mo/ugrid-checks/compare/main...ioos_category_trial).

Thanks! My mind on this is shifting a bit. While It would be awesome to have `ugrid-checks` available as a compliance-checker plugin. I wonder how many times we actually check model outputs for compliance? there are  hand full of models out there and they are either already compliant, or checked once and patched for compliance. The scenario is quire different from observed data that requires checker for each new platform/instrument deployed.
 
> I would note that, when I discussed with another person (? @MathewBiddle or someone else ?) in context of the existing code, it seemed that the existing implementation was a more structured validation approach than the ugrid-checks code, which is arguably a bit ad-hoc in it's approach (it was written under some time constraint). I think you can see that in the ioos-interfacing draft code outlined above .

Indeed.

> I think it's also worth saying that, like the UGRID spec itself, the ugrid-checks code has been pretty much static for some years + gets very little attention, or at least little feedback. Which is not to say that it doesn't provide some value, and is still functional.

Agreed.

> So it does need careful consideration whether the somewhat hacky ugrid-checks code, and the above interfacing approach is generally good enough to adopt here -- I will appreciate comment on that from others !

TL;DR we will keep this plugin for legacy purposes, but I'm more inclined into recommend the very few people doing ugrid stuff to just use `ugrid-checks`.

Thanks for all the help and consideration @pp-mo! (And sorry for the long delay on getting back you on this.)
